Company Profile Page

CiscoPr opened the pull request:
Closes #86
Closes #196

CiscoPr committed:
added minimal things to company page

ttoino commented:
Ideally this would use the same route as in #48.

CiscoPr commented:
> Ideally this would use the same route as in #48.

Yeah, i will refactor that

CiscoPr committed:
deleted separate page of company

CiscoPr committed:
refactored profile page to be available to companies too

Naapperas commented:
Don't forget to format the code

CiscoPr committed:
very initial draft of the company page table for participants who interacted with them

pedronunes19 commented on resources/js/Pages/Profile/Show.vue:
```diff
@@ -23,41 +24,23 @@ defineProps<Props>();
+                        'App\\Models\\Company'
+                    "
+                    :item="user ?? $page.props.auth.user"
+                />
```
this should only appear when either user.usertype_type is Participant (if it exists) or the auth user is Participant

CiscoPr replied:
yeah, i forgot some cases for this condition, ty

CiscoPr committed:
corrected v-if and formated

CiscoPr committed:
added simple hasmanydeep

CiscoPr committed:
formatted

pedronunes19 committed:
defined gate for viewing user profiles

CiscoPr committed:
added query in user controller, array is still undefined

Naapperas commented:
What is missing? The student visit history?

CiscoPr commented:
> What is missing? The student visit history?

are you talking about the table?

Naapperas commented:
> > What is missing? The student visit history?
> 
> are you talking about the table?

I'm talking about what features or fixes are missing for this PR to be merged.

CiscoPr commented:
> > > What is missing? The student visit history?
> > 
> > 
> > are you talking about the table?
> 
> I'm talking about what features or fixes are missing for this PR to be merged.

what is missing is to get the participants correctly... the relationship as it is for now always returns an undefined array

Naapperas commented:
> > > > What is missing? The student visit history?
> > > 
> > > 
> > > are you talking about the table?
> > 
> > 
> > I'm talking about what features or fixes are missing for this PR to be merged.
> 
> what is missing is to get the participants correctly... the relationship as it is for now always returns an undefined array

Due to the way quests are linked to stands, we may need more complex logic than just having a deep query.

At what point does the query fail? That's the point where we need to act upon.

pedronunes19 committed:
add middleware to profile route

pedronunes19 commented on resources/js/Pages/Profile/Show.vue:
```diff
@@ -9,12 +9,15 @@ import TwoFactorAuthenticationForm from "@/Pages/Profile/Partials/TwoFactorAuthe
     sessions: Session[];
-    participant: Participant;
+    user: User;
+    participants: Participant[];
```
I don't think this prop is being passed when we use the default route from jetstream, that's likely why it's always undefined (one of the reasons at least)

CiscoPr replied:
You made a good point. However, i already had noted that... i just forgot to push :,))
And yet again, this doesn't solve the problem

pedronunes19 committed:
add authorization to profile page

pedronunes19 committed:
finish CV permissions

Naapperas committed:
Reworked 'visit history' query, reworked user controller, reworked user profile page

Naapperas committed:
Started validating company access to user pages

Naapperas committed:
Redid authorization logic for profile pages

Naapperas committed:
Added different tabs for different types of user

Naapperas committed:
Made it so companies can access profiles of students that have visited them

There are some errors when going to the company profile after visiting the uparticipant profile

Naapperas committed:
Added fallback for companies without visits

Naapperas committed:
Reworked program tiers

Naapperas committed:
Small visual changes

Naapperas committed:
Added dynamic stands for each day in an event

Naapperas committed:
Fixed sponsor color problem

Naapperas committed:
Fixed error with repeated sponsor tiers

Naapperas committed:
Stylized and added some final touches to visit history table

toni-santos commented on app/Http/Controllers/UserController.php:
```diff
@@ -15,41 +19,126 @@ public function show(Request $request)
+            if ($requestUser->cannot('viewProfileOf', $paramUser)) {
+                // If the currently logged in user is not an admin, they can only view their own profile (or other users if they are a company and the user is a participant of that company)
+
+                return redirect()->route('profile.show')->dangerBanner('Não tens acesso a esta página.');
```
Why do this instead of just showing 403?

Naapperas replied:
:boom: error page :boom:

Naapperas replied:
I'm marking this as resolved but if you feel that this should be changed go ahead.

toni-santos commented on app/Providers/AuthServiceProvider.php:
```diff
@@ -68,5 +68,35 @@ public function boot(): void
+        Gate::define('viewCVOf', fn (User $user, User $cv_user, Edition $edition) => (
+            ( // admins have access to all CVs that do not belong to admins (since they do not have those)
+                $user->isAdmin() &&
+                ! $cv_user->isAdmin()
```
What do you mean by "that do not belong to admins"?

Naapperas replied:
Perhaps my wording was off here: this was just a way of securing this validation agains an admin trying to view another admin's profile.

CiscoPr commented:
we should be careful, because, when clicking on this on a student profile as a company, it will return this error:
![image](https://github.com/semanadeinformatica/website-2023/assets/79597450/0120ef1a-7307-45a9-b70b-ff0f6fa5aa27)
![image](https://github.com/semanadeinformatica/website-2023/assets/79597450/2188c08b-1e77-4da7-88c6-017b7739dd2a)

CiscoPr commented:
on another note, we should also increase the width of this CV button in the same situation (company visiting enrolled student page)
![image](https://github.com/semanadeinformatica/website-2023/assets/79597450/c03213c8-8714-48af-bb64-e4eb9e214965)

CiscoPr commented:
Finally, i think we should also decrease the size of the "no students enrolled yet" message. Imo, it's too big

CiscoPr committed:
corrected size of text and position

pedronunes19 commented:
> we should be careful, because, when clicking on this on a student profile as a company, it will return this error: ![image](https://user-images.githubusercontent.com/79597450/274666274-0120ef1a-7307-45a9-b70b-ff0f6fa5aa27.png) ![image](https://user-images.githubusercontent.com/79597450/274666339-2188c08b-1e77-4da7-88c6-017b7739dd2a.png)

Why can a company even see this?

Naapperas commented:
> > we should be careful, because, when clicking on this on a student profile as a company, it will return this error: ![image](https://user-images.githubusercontent.com/79597450/274666274-0120ef1a-7307-45a9-b70b-ff0f6fa5aa27.png) ![image](https://user-images.githubusercontent.com/79597450/274666339-2188c08b-1e77-4da7-88c6-017b7739dd2a.png)
> 
> Why can a company even see this?

I believe companies should at least be able to see what events the user has joined that may be related to them

Naapperas committed:
Fixed some errors

Naapperas committed:
Made it so companies cannot see user events on participant profile, made it so CVs are already opened if the visitor is a company

Naapperas committed:
Fixed program timeline actions not showing associated users

Naapperas committed:
Fixed error when accessing Program page as an unathenticated user
